### PR TITLE
Honor RAnalHints and optimize aac (5-9 times faster)

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

File: libr/core/cconfig.c [hunk 1]
```diff
@@ -566,7 +566,13 @@ static int cb_asmbits(void *user, void *data) {
 	}
 
 	bits = node->i_value;
-
+#if 0
```
Comment (ret2libc) [inline libr/core/cconfig.c]: what about removing this? code is probably not that useful since it breaks stuff... if it worked it wouldn't have been commented out

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -3103,6 +3103,7 @@ void cmd_anal_reg(RCore *core, const char *str) {
 			free (ostr);
 			return;
 		}
+		// if !arg
```
Comment (ret2libc) [inline libr/core/cmd_anal.c]: remove comment

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -4691,64 +4692,69 @@ static void cmd_anal_blocks(RCore *core, const char *input) {
...
+	ut8 *block1;
+	int bufi = 0;
+	int bufi_max = bsz - 16;
 	if (addr_end - addr > UT32_MAX) {
 		return;
 	}
 	buf = malloc (bsz);
-	block = malloc (bsz);
-	if (!buf || !block) {
+	block0 = malloc (bsz);
```
Comment (ret2libc) [inline libr/core/cmd_anal.c]: just `calloc` here? instead of doing `malloc` + `memset(.., 0, ..)`

File: libr/core/cmd_anal.c [hunk 4]
```diff
@@ -4691,64 +4692,69 @@ static void cmd_anal_blocks(RCore *core, const char *input) {
...
 			break;
 		}
 		// TODO: too many ioreads here
-		if (bufi > 4000) {
+		if (bufi > bufi_max) {
 			bufi = 0;
 		}
 		if (!bufi) {
-			r_io_read_at (core->io, addr, buf, bsz);
+			memset (buf, 0, bsz);
```
Comment (ret2libc) [inline libr/core/cmd_anal.c]: why is this necessary? it seems to me `r_io_read_at` already does the `memset` with the io->ff value when necessary.

File: libr/core/cconfig.c (hunk 1 above)
Comment (radare) [reply libr/core/cconfig.c]: it breaks stuff for a reason that must be investigated i prefer to keep the comment there for future review

File: libr/core/cmd_anal.c (hunk 4 above)
Comment (radare) [reply libr/core/cmd_anal.c]: removed the memset

File: libr/core/cmd_anal.c (hunk 3 above)
Comment (radare) [reply libr/core/cmd_anal.c]: done

File: libr/core/cmd_anal.c (hunk 2 above)
Comment (radare) [reply libr/core/cmd_anal.c]: meh
